kedro/io/data_catalog.py

```diff
@@ -567,13 +638,42 @@ def list(self, regex_search: str | None = None) -> list[str]:
             ) from exc
         return [dset_name for dset_name in self._data_sets if pattern.search(dset_name)]
 
+    def exists_in_catalog(self, dataset_name: str) -> bool:
+        """Check if a dataset exists in the catalog as an exact match or if it matches a pattern."""
+        if (
+            dataset_name in self._data_sets
+            or dataset_name in self._pattern_name_matches_cache
+        ):
+            return True
+        matched_pattern = self.match_name_against_pattern(dataset_name)
+        if self.dataset_patterns and matched_pattern:
```

Comment:
Is it necessary to check `if self.dataset_patterns`? If that it `False` then it shouldn't be possible to have a `matched_pattern` in the line above right?

Reply:
updated